circuits/src/zk_gadgets/comparators.rs

```diff
@@ -288,6 +286,114 @@ impl<const D: usize> LessThanGadget<D> {
 // | Multiprover Gadgets |
 // -----------------------
 
+/// A multiprover version of the equal gadget
+pub struct MultiproverEqGadget;
+impl MultiproverEqGadget {
+    /// Computes whether the given input is equal to zero
+    ///
+    /// Warning: This requires opening the value to both parties so that they
```

Comment:
in order to make this more clear to the caller, should we require that the value be opened outside of the gadget?